src/Module.Client/GUI/CrpgMissionGauntletMainAgentCheerControllerView.cs

```diff
@@ -0,0 +1,369 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Text;
+using Crpg.Module.Helpers;
+using NetworkMessages.FromClient;
+using TaleWorlds.Core;
+using TaleWorlds.Diamond;
+using TaleWorlds.Engine;
+using TaleWorlds.Engine.GauntletUI;
+using TaleWorlds.GauntletUI;
+using TaleWorlds.InputSystem;
+using TaleWorlds.Library;
+using TaleWorlds.Localization;
+using TaleWorlds.MountAndBlade;
+using TaleWorlds.MountAndBlade.GauntletUI.Mission;
+using TaleWorlds.MountAndBlade.View;
+using TaleWorlds.MountAndBlade.View.MissionViews;
+using TaleWorlds.MountAndBlade.ViewModelCollection;
+using TaleWorlds.MountAndBlade.ViewModelCollection.HUD;
+
+namespace Crpg.Module.GUI;
+
+[OverrideView(typeof(MissionMainAgentCheerBarkControllerView))]
+public class CrpgMissionGauntletMainAgentCheerControllerView : MissionView
+{
+    private readonly IMissionScreen _missionScreenAsInterface;
+
+    private readonly TextObject _cooldownInfoText = new("{=aogZyZlR}You need to wait {SECONDS} seconds until you can cheer/shout again.");
+
+    private readonly TextObject _barkCooldownTimeInfoText = new("{=BorZeZPs}You need to wait {SECONDS} seconds until you can shout again.");
+
+    private bool _holdHandled;
+
+    private float _holdTime;
+
+    private bool _prevCheerKeyDown;
+
+    private GauntletLayer? _gauntletLayer;
+
+    private MissionMainAgentCheerBarkControllerVM? _dataSource;
+
+    private float _cooldownTimeRemaining;
+
+    private int _barkUsesLeft;
+
+    private float _barkUsesResetCooldownTime;
+
+    private bool _isSelectingFromInput;
+
```

Comment:
Why so much whitespace!

Reply:
Ah, forgot about that one.
Was just copy pasted decompiled code and never gave another tought about that.